types/digitalTechnologies/humidity/uplink.js

```diff
@@ -0,0 +1,23 @@
+function consume(event) {
+  const { eventType } = event.data.event;
+  const sample = {};
+  let topic = eventType;
+
+  if (eventType === "humidity") {
+    sample.temperature = event.data.event.data.humidity.temperature;
+    sample.humidity = event.data.event.data.humidity.relativeHumidity;
+  } else if (eventType === "touch") {
+    sample.touch = true;
+  } else if (eventType === "networkStatus") {
+    sample.signalStrength = event.data.event.data.networkStatus.signalStrength;
+    sample.rssi = event.data.event.data.networkStatus.rssi;
+    sample.transmissionMode =
+      event.data.event.data.networkStatus.transmissionMode;
```

Comment:
why is the input so strange? - event.data.event.data?

Reply:
it's packaged by us into event.data and packaged by digital technologies into event.data etc

Reply:
@giepetto ok we should look into this, I guess we should directly forward event.data without having it twice